community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -88,10 +88,11 @@ public boolean nodeAddLabel( KernelStatement state, long nodeId, int labelId )
         {
             PropertyConstraint constraint = constraints.next();
             int propertyKeyId = constraint.propertyKeyId();
-            Property property = entityReadOperations.nodeGetProperty( state, nodeId, propertyKeyId );
-            if ( property.isDefined() )
+            Object value = entityReadOperations.nodeGetProperty( state, nodeId, propertyKeyId );
+            if ( value != null )
             {
-                validateNoExistingNodeWithLabelAndProperty( state, labelId, (DefinedProperty) property, nodeId );
+                DefinedProperty property = Property.property( propertyKeyId, value );
```

Comment:
I think we can avoid creation of `DefinedProperty` and modify `validateNoExistingNodeWithLabelAndProperty` to accept `propertyKeyId` and `value` separately.